lib/takeaway.rb

```diff
@@ -1,38 +1,36 @@
 require_relative 'menu'
-require_relative 'order'
 require_relative 'textmessage'
 
-
 class Takeaway
 
   attr_reader :order, :order_total
 
-def initialize
-  @menu = Menu.new
-  @order = []
-  @order_total = 0
-end
+  def initialize
+    @menu = Menu.new
+    @order = []
+    @order_total = 0
+  end
 
-def view_menu
-  @menu.dishes
-end
+  def view_menu
+    @menu.dishes
+  end
 
-def add_to_order(dish)
-  @order << dish
-end
+  def add_to_order(dish)
+    @order << dish
+  end
 
-def calculated_total
+  def calculated_total
 
-  @order.each do |dish|
-    @order_total += @menu.dishes[dish]
+    @order.each do |dish|
+      @order_total += @menu.dishes[dish]
+    end
+    @order_total
   end
-  @order_total
-end
 
-def confirm_order(total)
-  raise("That is not the correct total!") if total != @order_total
-text = TextMessage.new
-text.send_text
-end
+  def confirm_order(total)
```

Comment:
this makes sense according to challenge description, so less an issue with you and more an issue with how they wrote it. But I think confirm_order would either be a function of the actual takeaway system (i.e. their system processes the order_total), or the user would simply be given the order_total automatically underneath a list of individual orders with their respective amounts and prices. That's more complicated though, so this is a good start.